### PR TITLE
[ADAM-1053] Clean up Transform

### LINKED ISSUE #1053: Clean up if clauses in Transform
`org.bdgenomics.adam.cli.Transform` has a ton of if clauses that allow us to treat the `Transform` class essentially as a one-stop-shop for all your Read Transforming Needs™.

However, as has been [rightly pointed out](https://github.com/bigdatagenomics/adam/pull/1045#discussion_r66105933), this is a mess to read and follow. As such, we should clean up the tangled web of transforms in `Transform`.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1053. Split out into many commits for reviewer convenience, but also to make it easier to address merge conflicts when #1011 merges. Will squash before merge.

### CODE REVIEW

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala [hunk 1]
```diff
@@ -166,74 +194,175 @@ class Transform(protected val args: TransformArgs) extends BDGSparkCommand[Trans
...
+        fragmentLength = args.mdTagsFragmentSize,
+        overwriteExistingTags = args.mdTagsOverwrite,
+        validationStringency = stringencyOpt.getOrElse(ValidationStringency.STRICT)
+      )
+    } else {
+      rdd
     }
+  }
+
+  def apply(rdd: RDD[AlignmentRecord],
```
Comment (heuermh) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: this cleans up nice!

Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/Transform.scala]: Agreed. I was quite pleased at the end. I was inspired by @ryan-williams's `maybeX` pattern, which we've got elsewhere.

**Comment by heuermh** [comment]
Will let you rebase and resolve conflicts, then I'll give a review pass.

**Comment by fnothaft** [comment]
Rebased!

**Comment by heuermh** [comment]
LGTM

Squash then I'll merge.

**Comment by fnothaft** [comment]
Squashed!

**Comment by heuermh** [comment]
Merged commit c251f79c6bde3dce12e685c6cf03d5b1c30e9273.

Thank you, @fnothaft!

**Pull request merged**